backend/de.metas.business/src/main/java/de/metas/invoice/service/impl/AbstractInvoiceBL.java

```diff
@@ -1746,4 +1746,40 @@ public PaymentRule getDefaultPaymentRule()
 		return sysconfigs.getReferenceListAware(SYSCONFIG_C_Invoice_PaymentRule, PaymentRule.OnCredit, PaymentRule.class);
 	}
 
+	@Override
+	public final void discountInvoice(final org.compiere.model.I_C_Invoice invoice, final BigDecimal discountAmt, final String description)
+	{
+		if (discountAmt.signum() == 0)
+		{
+			return;
+		}
+
+		final BigDecimal discountAmtAbs;
+		if (!invoice.isSOTrx())
+		{
+			// API
+			discountAmtAbs = discountAmt.negate();
+		}
+		else
+		{
+			// ARI
+			discountAmtAbs = discountAmt;
+		}
+
+		// @formatter:off
+		Services.get(IAllocationBL.class).newBuilder()
+			.orgId(invoice.getAD_Org_ID())
+			.currencyId(invoice.getC_Currency_ID())
+			.dateAcct(invoice.getDateAcct())
+			.dateTrx(invoice.getDateInvoiced())
```

Comment:
not sure if using invoice's DateAcct/DateInvoiced is correct.

Reply:
Need to clear it with Mark